driver/src/contracts/stablex_contract.rs

```diff
@@ -56,6 +56,13 @@ impl StableXContractImpl {
     }
 }
 
+pub struct FilteredAuctionData {
```

Comment:
naming nit: Maybe include in the name somehow that this is a single page of auction data? I'm fine with how it is though.